central/notifiers/jira/jira.go

```diff
@@ -69,57 +59,178 @@ type jira struct {
 }
 
 type issueTypeResult struct {
-	StartAt    int                      `json:"startAt"`
-	MaxResults int                      `json:"maxResults"`
-	Total      int                      `json:"total"`
-	IssueTypes []*jiraLib.MetaIssueType `json:"values"`
+	StartAt         int                      `json:"startAt"`
+	MaxResults      int                      `json:"maxResults"`
+	Total           int                      `json:"total"`
+	IssueTypes      []*jiraLib.MetaIssueType `json:"values"`
+	IssueTypesCloud []*jiraLib.MetaIssueType `json:"issueTypes"`
 }
 
-func getIssueTypes(client *jiraLib.Client, project string) ([]*jiraLib.MetaIssueType, error) {
-	// Low level HTTP client call is used here due to the deprecation/removal of the Jira endpoint used by the Jira library
-	// to fetch the CreateMeta data, and there is no API call for the suggested endpoint to use in place of the removed one.
-	// Info here:
-	// https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html
-	urlPath := fmt.Sprintf("rest/api/2/issue/createmeta/%s/issuetypes", project)
+type issueField struct {
+	Name    string `json:"name"`
+	Key     string `json:"key"`
+	FieldID string `json:"fieldId"`
+}
+
+type issueFieldsResult struct {
+	StartAt          int           `json:"startAt"`
+	MaxResults       int           `json:"maxResults"`
+	Total            int           `json:"total"`
+	IssueFields      []*issueField `json:"values"`
+	IssueFieldsCloud []*issueField `json:"fields"`
+}
+
+type permissionResult struct {
+	Permissions map[string]struct {
+		HavePermission bool
+	}
+}
+
+func callJira(client *jiraLib.Client, urlPath string, result interface{}, startAt int) error {
+	if strings.Contains(urlPath, "?") {
+		urlPath = urlPath + fmt.Sprintf("&startAt=%d", startAt)
+	} else {
+		urlPath = urlPath + fmt.Sprintf("?startAt=%d", startAt)
+	}
 
 	req, err := client.NewRequest("GET", urlPath, nil)
 	if err != nil {
-		return []*jiraLib.MetaIssueType{}, err
+		return err
```

Comment:
wrap these errors with what we were doing. e.g. couldn't create request and below, failed to call jira or something like that